xio-core/src/main/java/com/xjeffrose/xio/http/RoutePartial.java

```diff
@@ -1,22 +1,26 @@
 package com.xjeffrose.xio.http;
 
 import io.netty.channel.ChannelHandlerContext;
+import lombok.Getter;
 import lombok.ToString;
+import lombok.experimental.Accessors;
 
 @ToString
+@Accessors(fluent = true)
+@Getter
 public class RoutePartial {
 
-  protected final Request request;
-  protected final Route route;
-  protected final PipelineRequestHandler handler;
+  private final Request request;
+  private final String path;
+  private final RouteState route;
 
-  RoutePartial(Request request, Route route, PipelineRequestHandler handler) {
+  RoutePartial(Request request, String path, RouteState route) {
     this.request = request;
+    this.path = path;
     this.route = route;
-    this.handler = handler;
   }
 
   void apply(ChannelHandlerContext ctx) {
-    handler.handle(ctx, request, route);
+    route.handler().handle(ctx, request, route);
```

Comment:
Is handler() guaranteed to not be null here? Does lombok take care of this for us?

Reply:
No, this is a java feature: https://github.com/xjdr/xio/blob/9391ca706467d3a88c71e8a3dd557dd4a140bd0d/xio-core/src/main/java/com/xjeffrose/xio/http/RouteState.java#L13

https://en.wikipedia.org/wiki/Final_(Java)#Final_variables

Technically the caller to `new RouteState` could pass in null, but that would be programmer error. We could put some precondition checks into the RouteState constructor but I generally like to avoid that as it makes unit tests very verbose when you're not allowed to pass in a null value. 

The flipside to that is using a builder pattern in your unit tests so that the state object is constructed correctly.

I could go either way here.